pkg/importer/dataStream.go

```diff
@@ -703,6 +714,51 @@ func copy(r io.Reader, out string, qemu bool, imageSize string) error {
 	return nil
 }
 
+//ValidateSpaceConstraint - validates wheather there is enough space in PVC for both qcow2 image and converted raw image
+//Assumes it is possible to retrieve info from qcow2 file by means of qemu-img utility
+//Returns failure if either of the following happens
+//1. info cannot be retrieved from qcow2 file
+//2. Either ActualSize or VirtualSize are 0 - not specifed
+//3. ActualSize and VirtualSize together exceed available PVC Space
+func ValidateSpaceConstraint(qcow2Image string, destDir string) (bool, error) {
+
+	//Verify there is enough space in pvc before conversion
+	info, err := qemuOperations.Info(qcow2Image)
+
+	if err != nil {
+		return false, errors.Wrap(err, "Local image validation failed to retrieve image info")
+	}
+
+	if info == nil {
+		return false, errors.Wrap(err, "Local image validation failed to retrieve image info")
+	}
+
+	imgVirtualSize, succeed := resource.NewQuantity(info.VirtualSize, resource.BinarySI).AsInt64()
```

Comment:
AFAIK info.VirtualSize is an int64 (according to the Struct info), so why are we converting to Quantity and then back to int64. I think we can skip that step can't we, or am I missing something?